bundles/org.openhab.core.io.transport.modbus/src/main/java/org/openhab/core/io/transport/modbus/internal/pooling/ModbusSlaveConnectionFactoryImpl.java

```diff
@@ -183,27 +186,34 @@ public ModbusSlaveConnection create(ModbusSlaveEndpoint endpoint) throws Excepti
     }
 
     @Override
-    public PooledObject<ModbusSlaveConnection> wrap(ModbusSlaveConnection connection) {
+    public PooledObject<@Nullable ModbusSlaveConnection> wrap(@Nullable ModbusSlaveConnection connection) {
         return new PooledConnection(connection);
     }
 
     @Override
-    public void destroyObject(ModbusSlaveEndpoint endpoint, @Nullable PooledObject<ModbusSlaveConnection> obj) {
+    public void destroyObject(ModbusSlaveEndpoint endpoint,
+            @Nullable PooledObject<@Nullable ModbusSlaveConnection> obj) {
         if (obj == null) {
             return;
         }
-        logger.trace("destroyObject for connection {} and endpoint {} -> closing the connection", obj.getObject(),
-                endpoint);
-        obj.getObject().resetConnection();
+        ModbusSlaveConnection connection = obj.getObject();
+        if (connection == null) {
+            return;
+        }
+        logger.trace("destroyObject for connection {} and endpoint {} -> closing the connection", connection, endpoint);
+        connection.resetConnection();
     }
 
     @Override
-    public void activateObject(ModbusSlaveEndpoint endpoint, @Nullable PooledObject<ModbusSlaveConnection> obj)
-            throws Exception {
+    public void activateObject(ModbusSlaveEndpoint endpoint,
```

Comment:
This method raised NPE as well, but it was caught by the `catch(Exception e)` block below... leading to message logged.